src/main/java/core/basesyntax/GetFigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class GetFigureSupplier extends GetFigure {
+    private static final int RANGE = 100;
+    private static final String DEFAULT_COLOR = "WHITE";
+    private static int index;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+    private String figureType;
+    private GetFigure figure;
+    private String figureColor;
+
+    public GetFigure getRandomFigure() {
+        index = random.nextInt(FigureType.values().length);
+        figureType = FigureType.values()[index].name();
+        figureColor = colorSupplier.getRandomColor();
+        switch (figureType) {
```

Comment:
that's the perfect opportunity to use enum, why would you willingly rely on strings instead

Reply:
I think it's a good idea to use ENUM and CASE for this task